File: leet code  Sheet1/c++/283. Move Zeroes.cpp
#include<iostream>
#include<vector>
using namespace std;
void moveZeroes(vector<int>& nums) {
    
    int move = 0;
    
    for (int i=0;i<nums.size();i++) {
    	
    	if (nums[i] != 0){
    		nums[move] = nums[i];
    		move++;
		}
	}
	for (int i=move;i<nums.size();i++) { // i = 3 4     i<5
		nums[i] = 0;
	}
	
	return nums;
}
int main(){
	
}

File: leet code  Sheet1/java/283. Move Zeroes.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Main {
    public static void main(String[] argv)
    {
        Scanner scanner = new Scanner(System.in);

    }

    public void moveZeroes(int[] nums) {
        int move = 0;

        for(int i=0;i<nums.length;i++){


            if (nums[i] != 0){
                nums[move] = nums[i];
                move++;
            }
        }

        for (int i =move ;i<nums.length;i++) {
            nums[i] = 0;
        }
    }
}
